### PR TITLE
StackClient: Make KonnectorCollection.find call instance of DocumentCollection

### PR DESCRIPTION
**Pull request by gregorylegarec**
Using `function.call` was not working as expected and was causing issues in Banks, we actually need a full instance of DocumentCollection here, to have access to other methods and properties like `this.toMangoOptions`, `this.indexes`, etc.

We actually need ~a full instance and not only the find()
method from DocumentCollection~ to extend DocumentCollection

### CODE REVIEW

**Comment by ptbrowne** [comment]
What was the problem exactly ?

**Comment by ptbrowne** [comment]
I think it would work if AppCollection extended DocumentCollection. I think it's a better solution.

**Comment by gregorylegarec** [comment]
It would work for sure but I am afraid of all the implications and possible regressions. 

I suggest to merge this fix and add an issue to deal with as soon as I have 30 minutes to test this.

**Comment by gregorylegarec** [comment]
@ptbrowne I updated with an extend. It seems to work.

File: packages/cozy-stack-client/src/AppCollection.js [hunk 1]
```diff
@@ -37,31 +35,6 @@ class AppCollection {
...
-    throw new Error('create() method is not available for applications')
-  }
-
-  async update() {
-    throw new Error('update() method is not available for applications')
-  }
-
-  async destroy() {
-    throw new Error('destroy() method is not available for applications')
-  }
```
Comment (drazik) [inline packages/cozy-stack-client/src/AppCollection.js]: `AppCollection` will use `find` from `DocumentCollection`, but also `get`, `create`, `update` and `destroy`. Maybe you want to keep these methods implementations that throw an error here?

File: packages/cozy-stack-client/src/KonnectorCollection.js [hunk 2]
```diff
@@ -3,8 +3,12 @@ import AppCollection from './AppCollection'
 export const KONNECTORS_DOCTYPE = 'io.cozy.konnectors'
 
 class KonnectorCollection extends AppCollection {
-  doctype = KONNECTORS_DOCTYPE
   endpoint = '/konnectors/'
```
Comment (drazik) [inline packages/cozy-stack-client/src/KonnectorCollection.js]: Nit: you can also put this in the constructor, so there is not a mix between class properties and initialization in the constructor.

File: packages/cozy-stack-client/src/AppCollection.js [hunk 3]
```diff
@@ -37,31 +35,6 @@ class AppCollection {
...
   }
-
-  find(...args) {
-    console.warn(
-      `find() method for doctype '${
-        this.doctype
-      }' relies internally on DocumentCollection.find() and may not fetch all expected data.`
-    )
-    return DocumentCollection.prototype.find.call(this, ...args)
-  }
```
Comment (drazik) [inline packages/cozy-stack-client/src/AppCollection.js]: You will lose the `console.warn`. If you want to keep it, you can use `super.find()`

File: packages/cozy-stack-client/src/AppCollection.js (hunk 1 above)
Comment (gregorylegarec) [reply packages/cozy-stack-client/src/AppCollection.js]: Mmmmh I am not sure what we really want here. But maybe we should keep throwing while we have not decided.

**Pull request merged**